addons/account/models/account_move_line.py

```diff
@@ -765,12 +792,120 @@ def _compute_cumulated_balance(self):
         for record in self:
             record.cumulated_balance = result[record.id]
 
+    def _search_is_open_at_date(self, operator, value):
+        def to_sql(model, alias, query):
+            recon_check_alias = query.make_alias(alias, 'exclude_full_recon')
+            query.add_join(
+                kind="LEFT JOIN",
+                alias=recon_check_alias,
+                table=SQL(
+                    """
+                    (
+                        SELECT DISTINCT apr.full_reconcile_id,
+                               TRUE AS has_future_partial
+                          FROM account_partial_reconcile apr
+                         WHERE apr.full_reconcile_id IS NOT NULL
+                           AND apr.max_date > %(recon_date_limit)s
+                    )
+                    """,
+                    recon_date_limit=value,
+                ),
+                condition=SQL(
+                    "%(recon_check_alias)s.full_reconcile_id = %(original_alias)s.full_reconcile_id",
+                    recon_check_alias=SQL.identifier(recon_check_alias),
+                    original_alias=SQL.identifier(alias),
+                ),
+            )
+            return SQL("(%(original_alias)s.full_reconcile_id IS NULL OR %(recon_check_alias)s.has_future_partial IS NOT NULL)",
+                recon_check_alias=SQL.identifier(recon_check_alias),
+                original_alias=SQL.identifier(alias),
+            )
+        return Domain.custom(to_sql=to_sql)
+
+    def _search_reconciled_at_date(self, operator, value):
+        if operator not in ('<', '<=', '='):
```

Comment:
should maybe add a comment mentioning that this only works with OR
eg. `['|', ('reconciled_at_date', '<=', '2025-12-31'), ('id', 'in', self.ids)]`